lib/grp.gi

```diff
@@ -460,8 +464,26 @@ InstallMethod( IsSolvableGroup,
     # compute the derived series of <G>
     S := DerivedSeriesOfGroup( G );
 
+    # set IsAbelian filter
+    if Length( S ) > 2 then
+      Assert(2, not IsAbelian(G));
+      SetIsAbelian(G, false);
+    fi;
+
     # the group is solvable if the derived series reaches the trivial group
-    return IsTrivial( S[ Length( S ) ] );
+    if IsTrivial( S[ Length( S ) ] ) then
+      # set IsAbelian filter
+      if Length( S ) <=2 then
+        Assert(2, IsAbelian(G));
+        SetIsAbelian(G, true);
+      fi;
+      return true;
+    else
+      # set IsAbelian filter
+      Assert(2, not IsAbelian(G));
+      SetIsAbelian(G, false);
+      return false;
+    fi;
     end );
```

Comment:
This seems correct and fine, but I found it a bit hard to read and verify. How about something like this instead?
```
isSolvable := IsTrivial( S[ Length( S ) ] );
isAbelian := isSolvable and Length( S ) <= 2;
Assert(2, IsAbelian(G) = isAbelian);
SetIsAbelian(G, isAbelian);
return isSolvable;
```
I find that a lot easier to understand and verify, and it reduces code duplication.